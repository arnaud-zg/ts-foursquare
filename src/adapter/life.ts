import { NAuth, NOAuth2 } from '../../types'

export const adaptGetAccessToken = (
  payload: NOAuth2.IResponse
): NAuth.IAuth => ({
  accessToken: payload.access_token,
})
