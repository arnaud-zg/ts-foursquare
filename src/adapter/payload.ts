import { NHttpStatuses } from 'http-response-status'
import { NPayload } from '../../types/payload.d'

export const adaptPayload = <TResponse>(
  payload: NPayload.IPayload<TResponse>
) =>
  !!payload.meta &&
  payload.meta.code === NHttpStatuses.ESuccess.OK &&
  payload.response
