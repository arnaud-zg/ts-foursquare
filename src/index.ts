require('isomorphic-fetch')

export * from './standalone'
export * from './utils'

export * from './DEPRECATED_actions'
export * from './DEPRECATED_reducers'
export * from './DEPRECATED_selectors'
