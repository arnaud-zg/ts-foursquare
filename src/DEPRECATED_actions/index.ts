import { TLifeAction } from './life'

export type TRootAction = TLifeAction

export * from './life'
