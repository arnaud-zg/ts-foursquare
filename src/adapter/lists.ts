import { NLists, NPayload } from '../../types'
import { adaptPayload } from './payload'
import { ECommonError } from '../constants/error'

export const adaptGetLists = (
  payload: NPayload.IPayload<NLists.IResponse>
): NLists.IList | null => {
  const response = adaptPayload<NLists.IResponse>(payload)

  if (!response || !response.list) {
    throw new Error(ECommonError.RESPONSE_NOT_VALID)
  }

  return response.list
}
