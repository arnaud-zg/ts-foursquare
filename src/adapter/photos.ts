import { NPayload, NPhotos } from '../../types'
import { ECommonError } from '../constants/error'
import { adaptPayload } from './payload'

export const adaptGetPhotosDetails = (
  payload: NPayload.IPayload<NPhotos.IResponse>
): NPhotos.IPhoto => {
  const response = adaptPayload<NPhotos.IResponse>(payload)

  if (!response || !response.photo) {
    throw new Error(ECommonError.RESPONSE_NOT_VALID)
  }

  return response.photo
}
