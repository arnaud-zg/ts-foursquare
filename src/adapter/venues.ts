import { NHttpStatuses } from 'http-response-status'
import { NPayload } from '../../types/payload.d'
import { NRecommendedPlaces } from '../../types/recommendedPlaces'
import { NVenue } from '../../types/venue.d'

export const adaptGetVenuesSearch = (
  payload: NPayload.IPayload<NVenue.IResponse>
): NVenue.IVenue[] =>
  !!payload.meta &&
  payload.meta.code === NHttpStatuses.ESuccess.OK &&
  !!payload.response &&
  !!payload.response.venues &&
  !!payload.response.venues.length
    ? payload.response.venues
    : []

export const adaptGetVenuesExplore = (
  payload: NPayload.IPayload<NRecommendedPlaces.IResponse>
): NRecommendedPlaces.IGroupItem[] =>
  !!payload.meta &&
  payload.meta.code === NHttpStatuses.ESuccess.OK &&
  !!payload.response &&
  !!payload.response.groups &&
  !!payload.response.groups.length &&
  !!payload.response.groups[0].items &&
  !!payload.response.groups[0].items.length
    ? payload.response.groups[0].items
    : []

export const adaptGetVenuesTrending = (
  payload: NPayload.IPayload<NVenue.IResponse>
): NVenue.IVenue[] => adaptGetVenuesSearch(payload)
