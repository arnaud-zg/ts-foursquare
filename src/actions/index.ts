import { TLifeAction } from './life'
import { TVenuesAction } from './venues'

export type TRootActionType = TVenuesAction | TLifeAction

export type TRootAction = TVenuesAction | TLifeAction

export * from './life'
export * from './venues'
