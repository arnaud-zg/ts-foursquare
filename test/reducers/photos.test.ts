import { getPhotosDetailsAsync } from '../../src/DEPRECATED_actions'
import { adaptGetPhotosDetails } from '../../src/adapter'
import {
  initialState as initialStatePhotos,
  photosReducer,
} from '../../src/DEPRECATED_reducers/photos'
import {
  initialState as initialStateRequest,
  requestReducer,
} from '../../src/DEPRECATED_reducers/request'
import {
  initialState as initialStateStatus,
  statusReducer,
} from '../../src/DEPRECATED_reducers/status'
import { payload as payloadGetPhotosDetails } from '../epics/__mocks__/getPhotosDetailsAsync.revolve'

const genericFailureError = new Error('An error occurred')
const genericCancelReason = 'An user cancelled this action'

describe('reducers/lists/getPhotosDetailsAsync', () => {
  test.each`
    scenario                                                          | action
    ${`should get state after action: getPhotosDetailsAsync.success`} | ${getPhotosDetailsAsync.success(adaptGetPhotosDetails(payloadGetPhotosDetails))}
    ${`should get state after action: getPhotosDetailsAsync.cancel`}  | ${getPhotosDetailsAsync.cancel(genericCancelReason)}
    ${`should get state after action: getPhotosDetailsAsync.failure`} | ${getPhotosDetailsAsync.failure(genericFailureError)}
  `('$scenario with action: $action', ({ action }) => {
    expect(photosReducer(initialStatePhotos, action)).toMatchSnapshot()
    expect(requestReducer(initialStateRequest, action)).toMatchSnapshot()
    expect(statusReducer(initialStateStatus, action)).toMatchSnapshot()
  })
})
