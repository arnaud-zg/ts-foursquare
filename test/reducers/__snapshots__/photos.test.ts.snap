// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/lists/getPhotosDetailsAsync should get state after action: getPhotosDetailsAsync.cancel with action: {"payload": "An user cancelled this action", "type": "GET_PHOTOS_DETAILS_CANCEL"} 1`] = `
Object {
  "entities": Object {},
}
`;

exports[`reducers/lists/getPhotosDetailsAsync should get state after action: getPhotosDetailsAsync.cancel with action: {"payload": "An user cancelled this action", "type": "GET_PHOTOS_DETAILS_CANCEL"} 2`] = `
Object {
  "getPhotosDetailsAsync": Object {
    "cancelReason": "An user cancelled this action",
    "entityIds": Array [],
  },
}
`;

exports[`reducers/lists/getPhotosDetailsAsync should get state after action: getPhotosDetailsAsync.cancel with action: {"payload": "An user cancelled this action", "type": "GET_PHOTOS_DETAILS_CANCEL"} 3`] = `
Object {
  "getPhotosDetailsAsync": Object {
    "hasError": false,
    "isCancelled": true,
    "isLoading": false,
  },
}
`;

exports[`reducers/lists/getPhotosDetailsAsync should get state after action: getPhotosDetailsAsync.failure with action: {"payload": [Error: An error occurred], "type": "GET_PHOTOS_DETAILS_FAILURE"} 1`] = `
Object {
  "entities": Object {},
}
`;

exports[`reducers/lists/getPhotosDetailsAsync should get state after action: getPhotosDetailsAsync.failure with action: {"payload": [Error: An error occurred], "type": "GET_PHOTOS_DETAILS_FAILURE"} 2`] = `
Object {
  "getPhotosDetailsAsync": Object {
    "entityIds": Array [],
    "error": [Error: An error occurred],
  },
}
`;

exports[`reducers/lists/getPhotosDetailsAsync should get state after action: getPhotosDetailsAsync.failure with action: {"payload": [Error: An error occurred], "type": "GET_PHOTOS_DETAILS_FAILURE"} 3`] = `
Object {
  "getPhotosDetailsAsync": Object {
    "hasError": true,
    "isCancelled": false,
    "isLoading": false,
  },
}
`;

exports[`reducers/lists/getPhotosDetailsAsync should get state after action: getPhotosDetailsAsync.success with action: {"payload": [Object], "type": "GET_PHOTOS_DETAILS_SUCCESS"} 1`] = `
Object {
  "entities": Object {
    "51e4151c498e60b5d17bc721": Object {
      "createdAt": 1373902108,
      "height": 612,
      "id": "51e4151c498e60b5d17bc721",
      "prefix": "https://igx.4sqi.net/img/general/",
      "source": Object {
        "name": "Instagram",
        "url": "http://instagram.com",
      },
      "suffix": "/1022386_wEd7VXnWzp5lajvRLtAPunoiufDSIq8PMEtEuIH2Tzg.jpg",
      "user": Object {
        "firstName": "John",
        "gender": "male",
        "id": "12345678",
        "lastName": "Doe",
        "photo": Object {
          "prefix": "https://igx.4sqi.net/img/user/",
          "suffix": "/YXNQ5O4TLYSNHMKY.jpg",
        },
      },
      "venue": Object {
        "categories": Array [
          Object {
            "icon": Object {
              "prefix": "https://ss3.4sqi.net/img/categories_v2/arts_entertainment/museum_art_",
              "suffix": ".png",
            },
            "id": "4bf58dd8d48988d18f941735",
            "name": "Art Museum",
            "pluralName": "Art Museums",
            "primary": true,
            "shortName": "Art Museum",
          },
        ],
        "contact": Object {},
        "id": "427c0500f964a52097211fe3",
        "location": Object {
          "address": "1000 5th Ave",
          "cc": "US",
          "city": "New York",
          "country": "United States",
          "crossStreet": "btwn E 80th & E 84th St",
          "formattedAddress": Array [
            "1000 5th Ave (btwn E 80th & E 84th St)",
            "New York, NY 10028",
            "United States",
          ],
          "lat": 40.779150129460504,
          "lng": -73.9627398125614,
          "postalCode": "10028",
          "state": "NY",
        },
        "name": "The Metropolitan Museum of Art (Metropolitan Museum of Art)",
      },
      "visibility": "public",
      "width": 612,
    },
  },
}
`;

exports[`reducers/lists/getPhotosDetailsAsync should get state after action: getPhotosDetailsAsync.success with action: {"payload": [Object], "type": "GET_PHOTOS_DETAILS_SUCCESS"} 2`] = `
Object {
  "getPhotosDetailsAsync": Object {
    "entityIds": Array [
      "51e4151c498e60b5d17bc721",
    ],
  },
}
`;

exports[`reducers/lists/getPhotosDetailsAsync should get state after action: getPhotosDetailsAsync.success with action: {"payload": [Object], "type": "GET_PHOTOS_DETAILS_SUCCESS"} 3`] = `
Object {
  "getPhotosDetailsAsync": Object {
    "hasError": false,
    "isCancelled": false,
    "isLoading": false,
  },
}
`;
